app/routes/product-details.$productSlug/route.tsx

```diff
@@ -14,13 +14,24 @@ import { ProductPrice } from '~/src/components/product-price/product-price';
 import { QuantityInput } from '~/src/components/quantity-input/quantity-input';
 import { ProductOption } from '~/src/components/product-option/product-option';
 import { ShareProductLinks } from '~/src/components/share-product-links/share-product-links';
+import type { GetStaticRoutes } from '@wixc3/define-remix-app';
 
 import styles from './route.module.scss';
 
 export const loader = ({ params, request }: LoaderFunctionArgs) => {
     return getProductDetailsRouteData(params.productSlug, request.url);
 };
 
+export const getStaticRoutes: GetStaticRoutes = async () => {
+    const products = await getEcomApi().getAllProducts();
```

Comment:
@olehrakwix now when `getEcomApi()` is gone, what should be used here?

@PeterShershov in which environment does `getStaticRoutes` run? Preview?

Reply:
in this particular case, please use:
```
const api = createApi(createWixClient());
```

Reply:
Updated.

@olehrakwix I don't like that from the user's POV there are two completely different ways to get the API instance: `createApi(createWixClient())` and `initializeEcomApi()`.

Let's discuss this.

Reply:
@alisey `getStaticRoutes` is being called on the server.
see https://github.com/wixplosives/codux-core/blob/98d7de6e9bb6e4975068cd6f3ea1860082903ea9/packages/define-remix-app/src/define-remix-app.tsx#L280